app/src/main/java/net/gsantner/markor/ui/FileInfoDialog.java

```diff
@@ -0,0 +1,90 @@
+package net.gsantner.markor.ui;
+
+import android.app.Dialog;
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.DialogFragment;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AlertDialog;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import net.gsantner.markor.R;
+import net.gsantner.markor.util.AppCast;
+import net.gsantner.markor.util.AppSettings;
+import net.gsantner.opoc.util.FileUtils;
+
+import java.io.File;
+
+public class FileInfoDialog extends DialogFragment {
+    public static final String EXTRA_FILEPATH = "EXTRA_FILEPATH";
+    public static final String FRAGMENT_TAG = "fileInfoDialog";
+
+    private Context activityContext;
+
+    public static FileInfoDialog newInstance(File sourceFile) {
+        FileInfoDialog dialog = new FileInfoDialog();
+        Bundle args = new Bundle();
+        args.putSerializable(EXTRA_FILEPATH, sourceFile);
+        dialog.setArguments(args);
+        return dialog;
+    }
+
+    @Override
+    public void onAttach(Context context) {
+        super.onAttach(context);
+        this.activityContext = context;
+    }
+
+    private EditText _newNameField;
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        final File file = (File) getArguments().getSerializable(EXTRA_FILEPATH);
+
+        LayoutInflater inflater = LayoutInflater.from(getActivity());
+        AlertDialog.Builder dialogBuilder = setUpDialog(file, inflater);
+        AlertDialog dialog = dialogBuilder.show();
+        dialog.getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
+
+        _newNameField = dialog.findViewById(R.id.new_name);
+        return dialog;
+    }
+
+    private AlertDialog.Builder setUpDialog(final File file, LayoutInflater inflater) {
+        View root;
+        AlertDialog.Builder dialogBuilder;
+        boolean darkTheme = AppSettings.get().isDarkThemeEnabled();
+        dialogBuilder = new AlertDialog.Builder(getActivity(), darkTheme ?
+                R.style.Theme_AppCompat_Dialog : R.style.Theme_AppCompat_Light_Dialog);
+        root = inflater.inflate(R.layout.ui__file_info_dialog, null);
+
+        dialogBuilder.setTitle(getResources().getString(R.string.info));
+        dialogBuilder.setView(root);
+
+        TextView textView = root.findViewById(R.id.ui__filesystem_item__description);
+        textView.setText(file.getName());
```

Comment:
@gsantner I just wanted to keep a separate FileInfo class and populate it in a separate class and avoid adding file logic to this Dialog class.

Reply:
@gsantner I didn't create it, I was just trying to find where the best place would be to add the file.Like /utils